lib/modules/manager/argocd/extract.ts

```diff
@@ -15,17 +15,66 @@ import { fileTestRegex } from './util';
 
 function createDependency(
   definition: ApplicationDefinition
-): PackageDependency | null {
-  let source: ApplicationSource;
+): Array<PackageDependency | null> {
```

Comment:
```suggestion
): PackageDependency[] {
```
As viceice has said do not use null in this is prevent lower in the stack ( `processAppSpec` )